### PR TITLE
Support select tablet syntax in new parser

### LINKED ISSUE #4072: Support select tablet syntax in new parser
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
2. INSERT INTO '....'
3. SELECT  '....'

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [x] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4072

### CODE REVIEW

**Review by Seaven — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 14 / 14 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowTableStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/TableRelation.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowTableStatusStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowColumnStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowVariablesStmt.java|1|1|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 1]
```diff
@@ -752,7 +753,13 @@ public ParseNode visitTableName(StarRocksParser.TableNameContext context) {
             partitionNames = (PartitionNames) visit(context.partitionNames());
         }
 
-        TableRelation tableRelation = new TableRelation(tableName, partitionNames);
+        List<Long> tabletIds = Lists.newArrayList();
+        if (context.tabletList() != null) {
+            tabletIds = context.tabletList().INTEGER_VALUE().stream().map(ParseTree::getText)
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:758]: What if tablet list is empty?

Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:758]: tabletList cannot be empty. In antlr, the "?" keyword will produce Null and non-NULL, and the "*" keyword is an empty List, or non-empty.

**Comment by HangyuanLiu** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 14 / 14 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|5|5|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowTableStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/ast/TableRelation.java|4|4|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowTableStatusStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowColumnStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowDbStmt.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/analysis/ShowVariablesStmt.java|1|1|100.00%|

**Review by murphyatwork — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**